### PR TITLE
`mod decode`: Make remaining `fn`s safe

### LINKED ISSUE #843: `decode.rs`: Unsafe cleanup
There are ~700 unsafe operations happening in `decode.rs`.

- [ ] Cleanup unsafe operations happening within functions.
- [ ] Mark any functions that can be made fully safe as such.
- [ ] Add `unsafe` blocks around any remaining unsafe operations along with standard safety documentation.
- [ ] Add safety documentation for functions that need to remain unsafe.
- [ ] Add `#![deny(unsafe_op_in_unsafe_fn)]` to file.

### PR DESCRIPTION
**Pull request by randomPoison**
* Closes #843.

### CODE REVIEW

File: src/picture.rs [hunk 1]
```diff
@@ -238,39 +239,45 @@ pub fn rav1d_picture_copy_props(
...
-        &c.allocator,
+        allocator,
     )?;
 
     rav1d_picture_copy_props(
         &mut p.p,
-        c.content_light.clone(),
-        c.mastering_display.clone(),
+        content_light.clone(),
+        mastering_display.clone(),
```
Comment (kkysen) [inline src/picture.rs:270]: Can we move the `.clone()`s to the caller and pass the `Arc`s by value?  `.clone()`ing in the caller is generally more flexible.

**Review by kkysen — APPROVED** [review]
LGTM, just had one suggestion about moving the `Arc::clone`s to the caller.

**Comment by kkysen** [comment]
Oh, could you fix the formatting thing, too?

File: src/picture.rs (hunk 1 above)
Comment (randomPoison) [reply src/picture.rs]: Sure, done.

**Review by kkysen — APPROVED** [review]

**Pull request merged**